### PR TITLE
refactor the `ansi` help page

### PR DESCRIPTION
**Pull request by amtoine**
# Description
i've always found the `ansi --help` extra usage hard to read and understand...
i decided to give it a shot today, so here is what i came up :yum: 

- make the extra usage structured with `nushell` tables
- make the examples clearer with variables and comments

one change that might appear strange is the following last two commits
```diff
diff --git a/crates/nu-command/src/platform/ansi/ansi_.rs b/crates/nu-command/src/platform/ansi/ansi_.rs
index 4746d27fa..ba3e597c4 100644
--- a/crates/nu-command/src/platform/ansi/ansi_.rs
+++ b/crates/nu-command/src/platform/ansi/ansi_.rs
@@ -507,10 +507,7 @@ impl Command for AnsiCommand {
 
     fn signature(&self) -> Signature {
         Signature::build("ansi")
-            .input_output_types(vec![
-                (Type::Nothing, Type::String),
-                (Type::List(Box::new(Type::String)), Type::String),
-            ])
+            .input_output_types(vec![(Type::Nothing, Type::String)])
             .optional(
                 "code",
                 SyntaxShape::Any,
```
`ansi` is never used on `list` inputs, as can be seen in the `Ansi.run` function: `_input: PipelineData` is never used.
this broke the tests (see [this action](https://github.com/nushell/nushell/actions/runs/4589552235/jobs/8104520078#step:4:1392)) for no real reason...

# User-Facing Changes
hopefully an easier to read `help ansi` page.

# Tests + Formatting
- :green_circle: `toolkit fmt`
- :green_circle: `toolkit clippy`
- :green_circle: `toolkit test`
- :green_circle: `toolkit test stdlib`

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

**Comment by amtoine** [comment]
i also have these two commands which list
- available short names
```bash
> ansi --list | find "_" -v | find NUSHELL | reject preview
╭───┬─────────┬────────────┬────────╮
│ # │  name   │ short name │  code  │
├───┼─────────┼────────────┼────────┤
│ 0 │ green   │ g          │ \e[32m │
│ 1 │ red     │ r          │ \e[31m │
│ 2 │ blue    │ u          │ \e[34m │
│ 3 │ black   │ b          │ \e[30m │
│ 4 │ yellow  │ y          │ \e[33m │
│ 5 │ purple  │ p          │ \e[35m │
│ 6 │ magenta │ m          │ \e[35m │
│ 7 │ cyan    │ c          │ \e[36m │
│ 8 │ white   │ w          │ \e[37m │
│ 9 │ default │ def        │ \e[39m │
╰───┴─────────┴────────────┴────────╯
```
- available `attr` for `ansi -e`
```bash
> ansi --list
| where name =~ '^attr_'
| reject preview
| str replace "attr_" "" name
╭───┬───────────┬────────────┬───────╮
│ # │   name    │ short name │ code  │
├───┼───────────┼────────────┼───────┤
│ 0 │ normal    │ n          │ \e[0m │
│ 1 │ bold      │ bo         │ \e[1m │
│ 2 │ dimmed    │ d          │ \e[2m │
│ 3 │ italic    │ i          │ \e[3m │
│ 4 │ underline │ u          │ \e[4m │
│ 5 │ blink     │ bl         │ \e[5m │
│ 6 │ hidden    │ h          │ \e[8m │
│ 7 │ strike    │ s          │ \e[9m │
╰───┴───────────┴────────────┴───────╯
```

i think these would be helpful to have in the help page, but not sure where to put them :thinking: 

any idea ? :yum:

**Comment by fdncred** [comment]
So, the answer is to remove the extra_usage? I'm not sure how I feel about that. It is an introduction to those who do not understand what ansi escape sequences are. At a minimum, we should link to the wikipedia article on ansi escape sequences, but I'd prefer to leave this introduction. Maybe reword or put it in nushell tables to make it look better, but not remove it entirely.

I'm interested in seeing how others feel.

**Comment by amtoine** [comment]
> So, the answer is to remove the extra_usage? I'm not sure how I feel about that. It is an introduction to those who do not understand what ansi escape sequences are.
>
i do not feel it's really an introduction to the ANSI sequences :thinking: 
it's more like examples and i find redundant with what is already inside the `ansi --list` table :open_mouth: 
=> hence the two commands i mention above, which reconstruct the tables in the extra usage by using `ansi --list`

> At a minimum, we should link to the wikipedia article on ansi escape sequences
>
this i 100% agree with :+1: 

> but I'd prefer to leave this introduction. Maybe reword or put it in nushell tables to make it look better, but not remove it entirely.
> 
i propose
- we put back the link to the ANSI page, because it's really a good entry point
- we add the `ansi --list` commands above
  - in the extra usage => there will be no colors, quite hard to follow i think?
  - in examples => these are "small" examples, but as we can compute them with `ansi`, they are part of the command and could be put as the first two examples?

> I'm interested in seeing how others feel.
>
:+1:

**Comment by amtoine** [comment]
i'm open to anything :relieved: 

i simply find the long extra usage too hard to read and follow...
that's how i feel about it of course :wink:

**Comment by amtoine** [comment]
@fdncred 

i've put back
- the comments about `\x1b[` and `\x1b]` not being required with `-e` and `-o` in 102740a24ef1ff42577497a135c611d7e8d78dc1
- the link to the wikipedia page in the extra usage in 8572be5a5e6fe64a3b3499e62de83bcbdaa5c5cc

i'm gonna test a bit the tables :yum:

**Comment by amtoine** [comment]
ihave the following examples
```
Examples:
  List the available ANSI colors
  > ansi --list | find "_" -v | find NUSHELL | reject preview

  List the available ANSI attributes
  > ansi --list
    | where name =~ '^attr_'
    | reject preview
    | str replace "attr_" "" name
```
or i have the following usage
```
Output ANSI codes to change color and style of text.

An introduction to what ANSI escape sequences are can be found in the
[*ANSI escape code*](https://en.wikipedia.org/wiki/ANSI_escape_code) Wikipedia page.

Escape sequences usual values:
╭────┬────────────┬────────┬────────┬─────────╮
│  # │    type    │ normal │ bright │  name   │
├────┼────────────┼────────┼────────┼─────────┤
│  0 │ foreground │     30 │     90 │ black   │
│  1 │ foreground │     31 │     91 │ red     │
│  2 │ foreground │     32 │     92 │ green   │
│  3 │ foreground │     33 │     93 │ yellow  │
│  4 │ foreground │     34 │     94 │ blue    │
│  5 │ foreground │     35 │     95 │ magenta │
│  6 │ foreground │     36 │     96 │ cyan    │
│  7 │ foreground │     37 │     97 │ white   │
│  8 │ foreground │     39 │        │ default │
│  9 │ background │     40 │    100 │ black   │
│ 10 │ background │     41 │    101 │ red     │
│ 11 │ background │     42 │    102 │ green   │
│ 12 │ background │     43 │    103 │ yellow  │
│ 13 │ background │     44 │    104 │ blue    │
│ 14 │ background │     45 │    105 │ magenta │
│ 15 │ background │     46 │    106 │ cyan    │
│ 16 │ background │     47 │    107 │ white   │
│ 17 │ background │     49 │        │ default │
╰────┴────────────┴────────┴────────┴─────────╯

Escape sequences attributes:
╭───┬────┬──────────────┬──────────────────────────────╮
│ # │ id │ abbreviation │         description          │
├───┼────┼──────────────┼──────────────────────────────┤
│ 0 │  0 │              │ reset / normal display       │
│ 1 │  1 │ b            │ bold or increased intensity  │
│ 2 │  2 │ d            │ faint or decreased intensity │
│ 3 │  3 │ i            │ italic on (non-mono font)    │
│ 4 │  4 │ u            │ underline on                 │
│ 5 │  5 │ l            │ slow blink on                │
│ 6 │  6 │              │ fast blink on                │
│ 7 │  7 │ r            │ reverse video on             │
│ 8 │  8 │ h            │ nondisplayed (invisible) on  │
│ 9 │  9 │ s            │ strike-through on            │
╰───┴────┴──────────────┴──────────────────────────────╯

Operating system commands:
╭───┬─────┬───────────────────────────────────────╮
│ # │ id  │              description              │
├───┼─────┼───────────────────────────────────────┤
│ 0 │   0 │ Set window title and icon name        │
│ 1 │   1 │ Set icon name                         │
│ 2 │   2 │ Set window title                      │
│ 3 │   4 │ Set/read color palette                │
│ 4 │   9 │ iTerm2 Grown notifications            │
│ 5 │  10 │ Set foreground color (x11 color spec) │
│ 6 │  11 │ Set background color (x11 color spec) │
│ 7 │ ... │ others                                │
╰───┴─────┴───────────────────────────────────────╯
```

**Comment by fdncred** [comment]
I like those tables.

**Comment by amtoine** [comment]
> I like those tables.
>
oh yeah, i agree the tables look great, even all white :open_mouth: 

they're back in f1f508a3b7d14c5e969bb3c2f902d9ec8de4a252 :+1:

**Comment by fdncred** [comment]
Looks better. I'm sure we'll wait until after the release.

**Comment by amtoine** [comment]
> Looks better.
>
glad it's better now :relieved: 

> I'm sure we'll wait until after the release.
>
looking at the remaining time before the release, that makes sense :wink:

**Comment by fdncred** [comment]
Thanks @amtoine!!

**Comment by amtoine** [comment]
> Thanks @amtoine!!
>
glad you like it :relieved:

**Comment by fdncred** [comment]
We may want to change this to an ansi escaped link. Other than that, it looks nice.
```
[*ANSI escape code*](https://en.wikipedia.org/wiki/ANSI_escape_code) Wikipedia page.
```

**Comment by amtoine** [comment]
@fdncred 
how would you write that in plain text?

i see some binary magic :thinking: 
```bash
>_ "https://en.wikipedia.org/wiki/ANSI_escape_code"
| ansi link --text "*ANSI escape code*"
| into binary
Length: 78 (0x4e) bytes | printable whitespace ascii_other non_ascii
00000000:   1b 5d 38 3b  3b 68 74 74  70 73 3a 2f  2f 65 6e 2e   •]8;;https://en.
00000010:   77 69 6b 69  70 65 64 69  61 2e 6f 72  67 2f 77 69   wikipedia.org/wi
00000020:   6b 69 2f 41  4e 53 49 5f  65 73 63 61  70 65 5f 63   ki/ANSI_escape_c
00000030:   6f 64 65 1b  5c 2a 41 4e  53 49 20 65  73 63 61 70   ode•\*ANSI escap
00000040:   65 20 63 6f  64 65 2a 1b  5d 38 3b 3b  1b 5c         e code*•]8;;•\
```

**Comment by fdncred** [comment]
If instead of `into binary`, you do `debug -r`, you should be able to see what it's doing. Or either look at the source code for `ansi link`.

**Comment by amtoine** [comment]
> If instead of `into binary`, you do `debug -r`, you should be able to see what it's doing. Or either look at the source code for `ansi link`.
>
oooh ok, so a bunch of escape sequences :+1: 

you're asking me to add
```
"\u{1b}]8;;https://en.wikipedia.org/wiki/ANSI_escape_code\u{1b}\\*ANSI escape code*\u{1b}]8;;\u{1b}\\"
```
in replacement of the current markdown link, right?

just to be sure about the request :wink:

**Comment by fdncred** [comment]
> in replacement of the current markdown link, right?

Yup, that's how you do links in ansi escape sequences. I'd also remove the `*` before and after `ANSI escape code.
